docs/react-v9/contributing/rfcs/shared/build-system/stop-styles-transforms.md

```diff
@@ -0,0 +1,162 @@
+
+- [microsoft/griffel#453](https://github.com/microsoft/griffel/issues/453) (_check this issue for more details_)
+- [microsoft/griffel#526](https://github.com/microsoft/griffel/issues/526)
+
+_TL;DR_: Griffel relies on the order of CSS classes, which can lead to clashes when multiple bundles are involved. See the simple example below:
+
+```html
+<!-- CURRENT STATE -->
+
+<!-- main bundle -->
+<style>
+  .order0 {
+    padding: 10px;
+  }
+  .order1 {
+    padding-left: 5px;
+  }
+</style>
+
+<!-- CDN bundle -->
+<style>
+  .order0 {
+    padding: 10px;
+  }
+</style>
+
+<!-- HTML -->
+
+<!--
+  🔴 We expect that a "div" below will have "padding-left: 5px", but instead, it has "padding: 10px".
+
+     The issue appears because the CDN bundle loads after the main bundle, resulting in style overrides. This occurs because ".order0" appears in both bundles, and CSS prioritizes the order of appearance. 
+  -->
+
+<div class="order0 order1"></div>
+```
+
+The single reliable solution is to prefix styles with a unique identifier. This prevents CSS rules from colliding, for example:
+
+```html
+<!-- PROPOSAL -->
+
+<!-- main bundle -->
+<style>
+  .main-order0 {
+    padding: 10px;
+  }
+  .main-order1 {
+    padding-left: 5px;
+  }
+</style>
+
+<!-- CDN bundle -->
+<style>
+  .order0 {
+    padding: 10px;
+  }
+</style>
+
+<!-- HTML -->
+
+<!--
+  ✅ Now the "div" below will have "padding-left: 5px" as CSS rules don't clash anymore.
+  -->
+
+<div class="main-order0 main-order1"></div>
+```
+
+## Problem statement
+
+Integrating prefixing into Griffel is relatively straightforward. However, we currently undergo a pre-processing step in `@fluentui/react-components` that modifies JavaScript code.
+
+<details>
+  <summary>Original code</summary>
+
+```ts
+// packages/react-components/react-menu/src/components/MenuDivider/useMenuDividerStyles.styles.ts
+// 📝 output is simpfied
+
+const useStyles = makeStyles({
+  root: {
+    ...shorthands.margin('4px', '-5px', '4px', '-5px'),
+  },
+});
+```
+
+</details>
+
+<details>
+  <summary>Pre-processed code</summary>
+
+```ts
+// @fluentui/react-menu/lib/components/MenuDivider/useMenuDividerStyles.styles.js
+// 📝 output is simpfied
+
+const useStyles = /*#__PURE__*/ __styles(
+  {
+    root: {
+      B6of3ja: 'fvjh0tl',
+      t21cq0: ['f1rnx978', 'f1q7jvqi'],
+    },
+  },
+  {
+    d: ['.fvjh0tl{margin-top:4px;}', '.f1rnx978{margin-right:-5px;}', '.f1q7jvqi{margin-left:-5px;}'],
+  },
+);
+```
+
+</details>
+
+Addressing prefixing for pre-processed styles presents a challenge, as they have already been transformed into CSS rules and classes.
+
+## Proposal
+
+The long-term solution is to discontinue the pre-processing of styles in `@fluentui/react-components` and transfer this responsibility to the consuming application.
```

Comment:
Might be good idea to avoid these kind of "decisions" until we have it discussed internally and have consensus on what should be actual long term solution .

Reply:
It boils down to two problems that are opposites to each other:
1. Shipping pre-compiled as default to make sure that it is as fast as possible without userland work
2. Integration of multi-bundles and renderers will conflict with each others' styles

With 1) there is at least something possible in userland that can resolve that in a straightforward way. For 2), there is nothing 'simple' that can be done in userland.

Reply:
The problem is that currently we have:
- Fluent UI + Fluent AI that ship with AOT
- All other partners' packages that ship without AOT

That creates a ridiculous situation when the expectation is use only CSS extraction, but you **have to** configure AOT anyway for your code and for `node_modules`, too. My expectation that uses need to configure a single thing that just works.

---

I am fine with excluding this part as a longterm solution, however after some experience with that I am leaning towards it.